src/event_handler.rs

```diff
@@ -508,151 +457,192 @@ pub async fn handle_event_logs(
-                            || error_string.contains("transaction underpriced"))
-                    {
+                    JobsTxnSendError::GasPriceLow => {
                         pending_txn_data.gas_price =
                             U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
                                 * pending_txn_data.gas_price
                                 / 100;
-
-                        sleep(Duration::from_millis(10)).await;
                         continue;
                     }
+                    _ => {
+                        sleep(Duration::from_millis(200)).await;
+                        continue;
+                    }
+                }
+            };
 
-                    // Break in case of other errors like "nonce too low" or "execution reverted" or "insufficient funds" etc.
-                    retry_txn = false;
-                    break;
-                };
+            // Wait for confirmation of the sent txn
+            let Ok(Some(_)) = pending_txn
+                .confirmations(1)
+                .interval(Duration::from_secs(1))
+                .await
+            else {
+                // If even this simple dummy txn fails to be mined then check whether the operator has injected a new gas account to send the txns
+                if pending_txn_data.http_rpc_client.address()
+                    != app_state
+                        .http_rpc_client
+                        .lock()
+                        .unwrap()
+                        .clone()
+                        .unwrap()
+                        .address()
+                {
+                    // Since the current nonce is not getting resolved for the old gas account,
+                    // send the remaining pending txns having the old gas account to the transaction sender
+                    // channel again for resending with proper nonce handling
+                    while !pending_txns.lock().unwrap().is_empty() {
+                        let txn = pending_txns.lock().unwrap().pop_front().unwrap();
+
+                        if txn.http_rpc_client.address()
+                            == pending_txn_data.http_rpc_client.address()
+                        {
+                            let _ = tx_sender.send(txn.txn_data).await;
+                        } else {
+                            pending_txns.lock().unwrap().push_front(txn);
+                            break;
+                        }
+                    }
 
-                // Wait for confirmation of the sent txn
-                let Ok(Some(_)) = pending_txn
-                    .confirmations(1)
-                    .interval(Duration::from_secs(1))
-                    .await
-                else {
-                    // Retry if the txn is not confirmed
+                    // Break the loop for the current nonce and wait for future pending txns with the new gas account
                     break;
```

Comment:
this break intended to break loop at L466 or L562? I think L466, but it is breaking L562.

Reply:
No It is supposed to break the loop at L562 itself for the dummy txn sending part.

Reply:
Ohh yeah, you are right, why would we want to break outer loop :sweat_smile: . That's right.